src/config/wmodules-config.c

```diff
@@ -92,14 +84,14 @@ int Read_WModule(const OS_XML *xml, xml_node *node, void *d1, void *d2)
 #endif
 #ifndef WIN32
     else if (!strcmp(node->values[0], WM_AWS_CONTEXT.name)) {
-        if (wm_aws_read(children, cur_wmodule, agent_cfg) < 0) {
+        if (wm_aws_read(xml, children, cur_wmodule) < 0) {
             OS_ClearNode(children);
             return OS_INVALID;
         }
     }
 #ifndef CLIENT
     else if (!strcmp(node->values[0], WM_VULNDETECTOR_CONTEXT.name)) {
-        if (wm_vulnerability_detector_read(xml, children, cur_wmodule) < 0) {
```

Comment:
This change produced this error when compiling the manager:

```
config/wmodules-config.c: In function ‘Read_WModule’:
config/wmodules-config.c:94:44: warning: passing argument 1 of ‘wm_vulnerability_detector_read’ from incompatible pointer type [-Wincompatible-pointer-types]
         if (wm_vulnerability_detector_read(children, cur_wmodule) < 0) {
                                            ^~~~~~~~
In file included from ./wazuh_modules/wmodules.h:64:0,
                 from config/wmodules-config.c:12:
./wazuh_modules/wm_vuln_detector.h:279:5: note: expected ‘const OS_XML * {aka const struct _OS_XML *}’ but argument is of type ‘xml_node ** {aka struct _xml_node **}’
 int wm_vulnerability_detector_read(const OS_XML *xml, xml_node **nodes, wmodule *module);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
config/wmodules-config.c:94:54: warning: passing argument 2 of ‘wm_vulnerability_detector_read’ from incompatible pointer type [-Wincompatible-pointer-types]
         if (wm_vulnerability_detector_read(children, cur_wmodule) < 0) {
                                                      ^~~~~~~~~~~
In file included from ./wazuh_modules/wmodules.h:64:0,
                 from config/wmodules-config.c:12:
./wazuh_modules/wm_vuln_detector.h:279:5: note: expected ‘xml_node ** {aka struct _xml_node **}’ but argument is of type ‘wmodule * {aka struct wmodule *}’
 int wm_vulnerability_detector_read(const OS_XML *xml, xml_node **nodes, wmodule *module);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
config/wmodules-config.c:94:13: error: too few arguments to function ‘wm_vulnerability_detector_read’
         if (wm_vulnerability_detector_read(children, cur_wmodule) < 0) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./wazuh_modules/wmodules.h:64:0,
                 from config/wmodules-config.c:12:
./wazuh_modules/wm_vuln_detector.h:279:5: note: declared here
 int wm_vulnerability_detector_read(const OS_XML *xml, xml_node **nodes, wmodule *module);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The previous version of this line, including the parameter `xml`, lets the manager compile.